azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/inbox_manager.py

```diff
@@ -78,3 +108,20 @@ def route_c2d_message(self, incoming_message):
         self.c2d_message_inbox._put(incoming_message)
         logger.info("C2D message sent to inbox")
         return True
```

Comment:
>return True [](start = 8, length = 11)

I feel funny about seeing return True in a lot of places and return False in only one for some reason.
I wonder if we should not return anything, and in the place where we return false, just rethrow the exception.

Feels kinda funny to generate internal exceptions only to end up using return codes.

Reply:
Yeah, I also am not thrilled by this.

The reason we return a boolean rather than throw an exception is that this code is called in the transport, and the transport currently has no way to handle an exception in the routing of data to inboxes. And we definitely don't want to raise unhandled exceptions from the transport thread. Addressing this would require changes both in the client and the transport, so I'd rather do that in a separate PR tbh.

---
In reply to: [266185373](https://github.com/Azure/azure-iot-sdk-python-preview/pull/61#discussion_r266185373) [](ancestors = 266185373)